### ISSUE TITLE
npm install fails

### ISSUE THREAD

**Issue by khaledhosny** [graphicore/librebarcode#12]
Terminal output:
```
$ npm install

> contextify@0.1.15 install /home/khaled/Development/librebarcode/node_modules/contextify
> node-gyp rebuild

make: Entering directory '/home/khaled/Development/librebarcode/node_modules/contextify/build'
  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc: In static member function ‘static v8::Local<v8::Context> ContextWrap::createV8Context(v8::Local<v8::Object>)’:
../src/contextify.cc:131:68: warning: ‘v8::Local<v8::Object> v8::Function::NewInstance() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                                    ^
In file included from /home/khaled/.node-gyp/9.2.0/include/node/v8.h:26:0,
                 from /home/khaled/.node-gyp/9.2.0/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/khaled/.node-gyp/9.2.0/include/node/v8.h:3848:52: note: declared here
   V8_DEPRECATED("Use maybe version", Local<Object> NewInstance() const);
                                                    ^
/home/khaled/.node-gyp/9.2.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc:150:16: error: ‘class v8::ObjectTemplate’ has no member named ‘SetAccessCheckCallbacks’; did you mean ‘SetAccessCheckCallback’?
         otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
                ^~~~~~~~~~~~~~~~~~~~~~~
                SetAccessCheckCallback
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyGetter(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Value>&)’:
../src/contextify.cc:182:80: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                                                ^
In file included from /home/khaled/.node-gyp/9.2.0/include/node/v8.h:26:0,
                 from /home/khaled/.node-gyp/9.2.0/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/khaled/.node-gyp/9.2.0/include/node/v8.h:3401:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^
/home/khaled/.node-gyp/9.2.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyQuery(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Integer>&)’:
../src/contextify.cc:209:67: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                                                   ^
In file included from /home/khaled/.node-gyp/9.2.0/include/node/v8.h:26:0,
                 from /home/khaled/.node-gyp/9.2.0/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/khaled/.node-gyp/9.2.0/include/node/v8.h:3401:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^
/home/khaled/.node-gyp/9.2.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/contextify.cc:210:71: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
             !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                                                       ^
In file included from /home/khaled/.node-gyp/9.2.0/include/node/v8.h:26:0,
                 from /home/khaled/.node-gyp/9.2.0/include/node/node.h:63,
                 from ../src/contextify.cc:1:
/home/khaled/.node-gyp/9.2.0/include/node/v8.h:3401:30: note: declared here
                 Local<Value> GetRealNamedProperty(Local<String> key));
                              ^
/home/khaled/.node-gyp/9.2.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
make: *** [contextify.target.mk:95: Release/obj.target/contextify/src/contextify.o] Error 1
make: Leaving directory '/home/khaled/Development/librebarcode/node_modules/contextify/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:159:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Linux 4.13.12-1-ARCH
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/khaled/Development/librebarcode/node_modules/contextify
gyp ERR! node -v v9.2.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/khaled/.npm/_logs/2017-11-18T16_00_37_106Z-debug.log
```

**Comment by khaledhosny** [comment]
Upgrading jsdom from 3 to 4 (current version is 11!) fixes the issue:
```diff
diff --git a/package.json b/package.json
index da274ac..25f3b08 100644
--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "commander": "git://github.com/graphicore/commander.js.git#atem-integration",
     "fontkit": "^1.7.6",
     "helmsman": "^1.0.3",
-    "jsdom": "~3",
+    "jsdom": "~4",
     "libxmljs": "~0.18.0",
     "lie": "^3.0.2",
     "requirejs": "^2.2.0"
```

**Comment by graphicore** [comment]
Thanks for the report! I wonder why this didn't happen to me.

**Comment by khaledhosny** [comment]
No idea, not much of a node user myself. May be I have a newer version of node or a newer compiler.

**Comment by khaledhosny** [comment]
This was fixed in cfee5aa44dd3dd3e0f968d6e2163aeb9b2e003ac.
